src/Bridges/Constraint/bridges/split_hyperrectangle.jl

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    SplitHyperRectangleBridge{T,G,F} <: Bridges.Constraint.AbstractBridge
+
+`SplitHyperRectangleBridge` implements the following reformulation:
+
+  * ``f(x) \\in \\textsf{HyperRectangle}(l, u)`` to
+    ``[f(x) - l; u - f(x)] \\in \\mathbb{R}_+``.
+
+## Source node
+
+`SplitHyperRectangleBridge` supports:
+
+  * `F` in [`MOI.HyperRectangle`](@ref)
+
+## Target nodes
+
+`SplitHyperRectangleBridge` creates:
+
+  * `G` in [`MOI.Nonnegatives`](@ref)
+"""
+mutable struct SplitHyperRectangleBridge{T,G,F} <: AbstractBridge
+    ci::MOI.ConstraintIndex{G,MOI.Nonnegatives}
+    func::F
+    set::MOI.HyperRectangle{T}
+end
+
+const SplitHyperRectangle{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{SplitHyperRectangleBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{SplitHyperRectangleBridge{T,G,F}},
+    model::MOI.ModelLike,
+    f::F,
+    s::MOI.HyperRectangle,
+) where {T,G,F}
+    scalars = MOI.Utilities.eachscalar(f)
+    lower = [
+        MOI.Utilities.operate(-, T, fi, s.lower[i]) for
+        (i, fi) in enumerate(scalars) if isfinite(s.lower[i])
+    ]
+    upper = [
+        MOI.Utilities.operate(-, T, s.upper[i], fi) for
+        (i, fi) in enumerate(scalars) if isfinite(s.upper[i])
+    ]
+    g = MOI.Utilities.operate(vcat, T, lower..., upper...)
```

Comment:
This will need to compile a new method for new sizes of rectangles. If the rectangle has 1000 entries, it will compile a method with 2000 arguments?

Reply:
True. I'll have another go.

Reply:
Done